### PR TITLE
Connection/external links accessibility

### LINKED ISSUE #35479: Jetpack Connection: identify external links for screen readers
External links in the Jetpack connection flow are not identified as such by SRs (it’s the case in the footer, too).

Issue reported in p1HpG7-qVW-p2.

Screenshot 2024-02-13 at 2 15 57 PM
Screenshot 2024-02-13 at 2 16 53 PM

---

1. The bare minimum is to ensure screen readers can identify external links. Following [this article](https://css-tricks.com/accessible-svgs/), this could probably be done by removing the `aria-hidden` attribute from the `svg`, adding a `role=img` attribute, and adding a title with the text screen readers should read.
Screenshot 2024-02-13 at 2 18 58 PM

2. If some of the links use the `ExternalLink` Gutenberg component, here's [an example of an accessible version](https://automattic.github.io/jetpack-storybook/?path=/docs/gutenberg_components-externallink--docs).

3. A good accessibility practice is to let users tailor their browsing experience, for instance, by not preventing zooming or by not preventing users from setting a font size. In the same logic, users should be able to decide when to open links in a new tab. By default, they have this choice. If we force links to open in a new tab, we remove this choice from them. There are only a few good reasons to do it, such as not wanting to lose the progress of a form. [This article](https://blog.pope.tech/2024/01/02/how-to-make-external-links-accessible/) explains it in more detail.

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Fixes https://github.com/Automattic/jetpack/issues/35479

## Proposed changes:
Following [pattern](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/external-link/index.tsx#L70) in core WordPress we are adding an accessible message to External links within connection flow.
Screenshot 2024-02-14 at 11 43 42 AM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Spin the local branch on a fresh JN site.
* Go through connection flow with a ScreenReader (eg VoiceOver on mac) and check that external links are read so.

### CODE REVIEW

File: projects/js-packages/components/components/button/index.tsx [hunk 1]
```diff
@@ -49,7 +50,15 @@ const Button = forwardRef< HTMLInputElement, ButtonProps >( ( props, ref ) => {
 
 	const externalIconSize = size === 'normal' ? 20 : 16;
 	const externalIcon = isExternalLink && (
-		<Icon size={ externalIconSize } icon={ external } className={ styles[ 'external-icon' ] } />
+		<>
+			<Icon size={ externalIconSize } icon={ external } className={ styles[ 'external-icon' ] } />
+			<VisuallyHidden as="span">
```
Comment (andrii-lysenko) [inline projects/js-packages/components/components/button/index.tsx:55]: I was a bit hesitant if we should add Translations and text in the core button, but it seems that this approach they use in core wordpress. I think it ok to have it here, especially if we want to make all external links accessible.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
The implementation looks good!

Some JS tests need fixing.

File: projects/js-packages/components/components/button/index.tsx (hunk 1 above)
Comment (monsieur-z) [reply projects/js-packages/components/components/button/index.tsx:55]: I think it's okay too. The text is generic enough, and I'd say it's not far-fetched to assume that the codebases using this package already import `__`.

**Comment by monsieur-z** [comment]
Would it make sense to create an `ExternalLink` component with this icon and text?

**Comment by andrii-lysenko** [comment]
@monsieur-z Actually, we might just use core Gutenberg `ExternalLink`. But I think we could have a separate task or even project, to revise the codebase and fix/refactor these inconsistencies. Or should we just do this within this task? What do you think?

**Comment by monsieur-z** [comment]
> @monsieur-z Actually, we might just use core Gutenberg `ExternalLink`. But I think we could have a separate task or even project, to revise the codebase and fix/refactor these inconsistencies. Or should we just do this within this task? What do you think?

I'd say let's ship this PR 🚀 

Since there's already a Gutenberg component, I agree it'd be overkill to create something new. I'd be interested in knowing why we didn't use the Gutenberg component: awareness, styles, or something else. If we can use the Gutenberg component and match the Jetpack interface without too many workarounds, maybe it's worth giving it a try.

**Review by monsieur-z — APPROVED** [review]
:shipit:

**Pull request merged**